app/src/main/java/org/fossasia/openevent/general/utils/ErrorUtils.kt

```diff
@@ -0,0 +1,122 @@
+package org.fossasia.openevent.general.utils
+
+import timber.log.Timber
+import org.json.JSONObject
+import org.fossasia.openevent.general.utils.HttpErrors.BAD_REQUEST
+import org.fossasia.openevent.general.utils.HttpErrors.CONFLICT
+import org.fossasia.openevent.general.utils.HttpErrors.FORBIDDEN
+import org.fossasia.openevent.general.utils.HttpErrors.METHOD_NOT_ALLOWED
+import org.fossasia.openevent.general.utils.HttpErrors.NOT_FOUND
+import org.fossasia.openevent.general.utils.HttpErrors.REQUEST_TIMEOUT
+import org.fossasia.openevent.general.utils.HttpErrors.UNAUTHORIZED
+import org.fossasia.openevent.general.utils.HttpErrors.UNPROCESSABLE_ENTITY
+import org.fossasia.openevent.general.utils.StringUtils.isEmpty
+import retrofit2.HttpException
+
+const val ERRORS = "errors"
+const val SOURCE = "source"
+const val POINTER = "pointer"
+const val DETAIL = "detail"
+const val TITLE = "title"
+
+const val POINTER_LENGTH = 3
+
+object ErrorUtils {
+
+    fun getMessage(throwable: Throwable): Error {
+        var error = Error()
+        if (throwable is HttpException) {
+            val errorCode = throwable.code()
+
+            error = if (errorCode == BAD_REQUEST ||
+                errorCode == UNAUTHORIZED ||
+                errorCode == FORBIDDEN ||
+                errorCode == NOT_FOUND ||
+                errorCode == METHOD_NOT_ALLOWED ||
+                errorCode == REQUEST_TIMEOUT) {
```

Comment:
Is there any problem using Kotlin when expression for this here, I think it will look cleaner

Reply:
Sorry, can't get it?

Reply:
https://kotlinlang.org/docs/reference/control-flow.html#when-expression

Reply:
updated